### DISCUSSION TITLE
Want try new/halfdone UI?

### DISCUSSION THREAD

**Discussion by Astilleros** [jomjol/AI-on-the-edge-device#1164] [General]
Hello, I like this project.
To better understand how it worked, I started to redo the web, I don't know if it would help the project.
I'll take 1 month at times and I think I'm halfway through.
I never really made any contributions and I don't know if this change would break things that I don't take into account.
It can be tested without changing the code (except the FileServer part, which changes it to json).
There are 2 files, the index.html and an app.js. For now about 150kB all bundled.
I put a link here, and if anyone wants to comment on something I would appreciate it.
Just save old index.html, and put the 2 files to try.
https://github.com/Astilleros/New-Ui-

**Comment by caco3** [comment]
I can't have a look on it right now as I am abroad right now, will test it in 2 weeks.

In general, your code looks clean, which is great. The current web UI is work in progress, we recently refactored and cleaned it up a lot, but it is still quite a mess.

It would need to be API compatible, so it can be simple a replacement (OTA would work with that).

You have now everything in 2 files. I would prefer to split it to smaller files to have a better overview, eg. css should be in a separate file and maybe the javascript could also be splitted into categories?

**Comment by Astilleros** [comment]
Hi, i have all separated but i do "manual" bundle for css and html and use "esbuild" for js module. Out:all in 2 files, web load in less than 1 sec.
I updated the link content. The 2 files are in the "/dist" folder.

**Reply by caco3** [reply]
ah, nice, didn't know that tool.

In the long run I would suggest to remove the dist files and auto generate them using Github actions. This way they are always consistent with the source. I can help you with that if needed.
We recently switched to Github actions for our build as well.

**Comment by caco3** [comment]
I fired my VPN connection and quickly deployed it on one of my devices.
I like the the current UI, but your approach looks more modern.
I personally don't like serif font on Web UI's, it looks too old fashion.

If @jomjol and @haverland also are open to a new Web UI, my suggestion would be as following:
1. You create a pull request to get it into our rolling branch (maybe move all your files to a folder `/web-ui`. The current web UI is in `/sd-card/html` but thats IMO not the right place anymore)
2. We could add Github actions to build your dist files and add them to the update zip files (see eg. https://github.com/jomjol/AI-on-the-edge-device/actions/runs/3217004162).
3. On the current web UI, we add a menu point `Switch to new Web UI (experimental)`. And you your web UI add a menu point to switch back. This way people could switch back and forward without breaking anything.

**Reply by ghost** [reply]
if we must not change the update-process, it should not change much. Alternative first in a feature-branch. So we can decide late, if we add it for the next release.

**Reply by Astilleros** [reply]
I don't know much about Github commands, if the case arises I will surely need help, thanks!

**Reply by ghost** [reply]
We will help you to integrate it. Minifying and zipping can we do in an GitHub action.

**Comment by jomjol** [comment]
I had a look on it and it looks more modern. Nice work.

BUT:
* currently all our documentation, FAQs, ... is done for the current UI --> we would need to redo a lot of documentation 
* I don't see things, that we could only do with a new interface.

What I appreciate is the small size of the html, that would improve the update and UX.

Therefore I'm currently not very open for a new design, but rather for the new backbone to increase speed.

Would that be transferable to the current style?

**Reply by Astilleros** [reply]
Current UI its done with js as text/javascript, not realy a module and cant do imports: for me hard to organize and dont repeat code. But all html css or js can be bundled, minimified and gziped to improve UX.

**Comment by Astilleros** [comment]
Hello, this interface is not ready, it is unfinished and untested. 
I just wanted to know what you thought, in case it's worth continuing. 
The css and fonts are just a test, spend more effort to unify the js and tidy/clean it up a bit (I learn more this way).

**Comment by caco3** [comment]
I think we are open for a new UI with cleaner/faster code as the current one has a lot of legacy code in it.
Please also keep in mind that there are other, higher prio issues that we need to solve in the near future, so a new UI has not very high priority on our side!
Also, like @jomjol mentioned, it would help us and the current users, if the structure of the pages would stay similar. And if/once we decide to switch to a new UI, we would welcome your contribution also on documentation side (wiki).
On any question, I am sure we are happy to support you.
